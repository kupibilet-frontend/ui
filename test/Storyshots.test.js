import initStoryshots from '@storybook/addon-storyshots'

import 'jest-styled-components'

initStoryshots()

// react-test-renderrer does not support .findDOMNode
jest.mock('react-onclickoutside', () => {
  const React = require('react')

  return Сomponent => (
    (props) => (
      <onClickOutside-Mock-Decorator>
        <Сomponent {...props} />
      </onClickOutside-Mock-Decorator>
    )
  )
})

jest.mock('react-dom', () => {
  const ReactDOM = jest.genMockFromModule('react-dom')

  ReactDOM.findDOMNode = () => null
  return ReactDOM
})

// matchMedia polyfill
window.matchMedia = window.matchMedia || (() => { return { matches: false, addListener: () => {}, removeListener: () => {}, }; });
