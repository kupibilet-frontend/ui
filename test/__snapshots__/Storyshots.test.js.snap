// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Box Defalut 1`] = `
<div
  className="hMQnet"
>
  <div
    className="hNYgbG"
  >
    <p>
      box content
    </p>
  </div>
</div>
`;

exports[`Storyshots Buttons Default 1`] = `
<div
  className="hMQnet"
>
  <button
    className="no-global jOONnJ"
    disabled={undefined}
    size="normal"
  >
    <span
      className="ieBWQc"
      size="normal"
    >
      Авиабилеты
    </span>
  </button>
</div>
`;

exports[`Storyshots Buttons Disabled 1`] = `
<div
  className="hMQnet"
>
  <button
    className="no-global cBbibc"
    disabled={true}
    size="normal"
  >
    <span
      className="ieBWQc"
      size="normal"
    >
      Cake is a lie
    </span>
  </button>
</div>
`;

exports[`Storyshots Buttons Icon-only button 1`] = `
<div
  className="hMQnet"
>
  <button
    className="no-global kKYUSd"
    disabled={undefined}
    size="large"
  >
    <span
      className="dOFMSX"
      size="large"
    >
      <svg
        className="hJDrrb"
        size="normal"
        style={undefined}
      >
        <use
          xlinkHref="#kb_carrot_monochrome"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`Storyshots Buttons Sizes 1`] = `
<div
  className="hMQnet"
>
  <button
    className="no-global czvUIu"
    disabled={undefined}
    size="large"
  >
    <span
      className="ieBWQc"
      size="large"
    >
      Бальшой
    </span>
  </button>
</div>
`;

exports[`Storyshots Buttons With icons 1`] = `
<div
  className="hMQnet"
>
  <button
    className="no-global dHnOvN"
    disabled={undefined}
    size="normal"
  >
    <span
      className="YhApM"
      size="normal"
    >
      <svg
        className="cdianR"
        size="normal"
        style={undefined}
      >
        <use
          xlinkHref="#kb_plane"
        />
      </svg>
    </span>
    <span
      className="ehcfDj"
      size="normal"
    >
      Авиабилеты
    </span>
    <span
      className="YhApM"
      size="normal"
    >
      <svg
        className="hJDrrb"
        size="normal"
        style={undefined}
      >
        <use
          xlinkHref="#kb_hand-baggage"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`Storyshots Checkbox default 1`] = `
<div
  className="hMQnet"
>
  <label
    className="eiMGFu"
    disabled={false}
  >
    <span
      checked={false}
      className="checkbox gJhUxw"
      disabled={false}
    >
      <input
        checked={false}
        className="cjpCUo"
        disabled={false}
        onChange={[Function]}
        type="checkbox"
      />
    </span>
    <span
      className="label-text cPjOWw"
      disabled={false}
    >
      Авиабилеты
    </span>
  </label>
</div>
`;

exports[`Storyshots Checkbox disabled 1`] = `
<div
  className="hMQnet"
>
  <label
    className="dsvJUH"
    disabled={true}
  >
    <span
      checked={false}
      className="checkbox fhciYu"
      disabled={true}
    >
      <input
        checked={false}
        className="cjpCUo"
        disabled={true}
        onChange={[Function]}
        type="checkbox"
      />
    </span>
    <span
      className="label-text faSGNd"
      disabled={true}
    >
      Авиабилеты
    </span>
  </label>
</div>
`;

exports[`Storyshots Collapse Accordion 1`] = `
<div
  className="hMQnet"
>
  <div
    style={
      Object {
        "width": "252px",
      }
    }
  >
    <div
      className="kb-collapse"
      style={undefined}
    >
      <div
        className="kb-collapse-item fatUsE"
        style={undefined}
      >
        <div
          aria-expanded={false}
          className="kb-collapse-header"
          onClick={[Function]}
          role="tab"
        >
          Foo
        </div>
      </div>
      <div
        className="kb-collapse-item fatUsE"
        style={undefined}
      >
        <div
          aria-expanded={false}
          className="kb-collapse-header"
          onClick={[Function]}
          role="tab"
        >
          Bar
        </div>
      </div>
      <div
        className="kb-collapse-item fatUsE"
        style={undefined}
      >
        <div
          aria-expanded={false}
          className="kb-collapse-header"
          onClick={[Function]}
          role="tab"
        >
          Mdaaaaa
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots Collapse Default 1`] = `
<div
  className="hMQnet"
>
  <div
    style={
      Object {
        "width": "252px",
      }
    }
  >
    <div
      className="kb-collapse"
      style={undefined}
    >
      <div
        className="kb-collapse-item fatUsE"
        style={undefined}
      >
        <div
          aria-expanded={false}
          className="kb-collapse-header"
          onClick={[Function]}
          role="tab"
        >
          Foo
        </div>
      </div>
      <div
        className="kb-collapse-item fatUsE"
        style={undefined}
      >
        <div
          aria-expanded={false}
          className="kb-collapse-header"
          onClick={[Function]}
          role="tab"
        >
          Bar
        </div>
      </div>
      <div
        className="kb-collapse-item fatUsE"
        style={undefined}
      >
        <div
          aria-expanded={false}
          className="kb-collapse-header"
          onClick={[Function]}
          role="tab"
        >
          Mdaaaaa
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots Collapse Default active panel 1`] = `
<div
  className="hMQnet"
>
  <div
    style={
      Object {
        "width": "252px",
      }
    }
  >
    <div
      className="kb-collapse"
      style={undefined}
    >
      <div
        className="kb-collapse-item kb-collapse-item-active fatUsE"
        style={undefined}
      >
        <div
          aria-expanded={true}
          className="kb-collapse-header"
          onClick={[Function]}
          role="tab"
        >
          Foo
        </div>
        <div
          className="kb-collapse-content kb-collapse-content-active"
          role="tabpanel"
        >
          <div
            className="kb-collapse-content-box"
          >
            
              A dog is a type of domesticated animal.
              Known for its loyalty and faithfulness,
              it can be found as a welcome guest in many households across the world.
            
          </div>
        </div>
      </div>
      <div
        className="kb-collapse-item fatUsE"
        style={undefined}
      >
        <div
          aria-expanded={false}
          className="kb-collapse-header"
          onClick={[Function]}
          role="tab"
        >
          Bar
        </div>
      </div>
      <div
        className="kb-collapse-item fatUsE"
        style={undefined}
      >
        <div
          aria-expanded={false}
          className="kb-collapse-header"
          onClick={[Function]}
          role="tab"
        >
          Mdaaaaa
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots DateRange DateRange 1`] = `
<div
  className="hMQnet"
>
  <div
    className="jStPWm"
  >
    <div
      className="DateInput--startDate--placeholder DateInput--endDate--placeholder"
    >
      <div
        className="DateRangePicker"
      >
        <div>
          <div
            className="DateRangePickerInput"
          >
            <div
              className="DateInput"
            >
              <input
                aria-describedby=""
                aria-label="Туда"
                autoComplete="off"
                className="DateInput__input"
                disabled={false}
                id="startDate"
                name="startDate"
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                placeholder="Туда"
                readOnly={false}
                required={false}
                type="text"
                value=""
              />
              
              <div
                className="DateInput__display-text"
              >
                Туда
              </div>
            </div>
            <div
              className="DateRangePickerInput__arrow"
            >
              <span />
            </div>
            <div
              className="DateInput"
            >
              <input
                aria-describedby=""
                aria-label="Обратно"
                autoComplete="off"
                className="DateInput__input"
                disabled={false}
                id="endDate"
                name="endDate"
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                placeholder="Обратно"
                readOnly={false}
                required={false}
                type="text"
                value=""
              />
              
              <div
                className="DateInput__display-text"
              >
                Обратно
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots FilterSection Default 1`] = `
<div
  className="hMQnet"
>
  <div
    style={
      Object {
        "width": 252,
      }
    }
  >
    <section
      className="hNYgbG"
    >
      <header
        className="jNGgbk"
      >
        <div>
          Пересадки
        </div>
        <div>
          Сбросить
        </div>
      </header>
      <div
        className="fKdGeq"
      >
        <div>
          <div>
            <label
              className="eiMGFu"
              disabled={false}
            >
              <span
                checked={false}
                className="checkbox gJhUxw"
                disabled={false}
              >
                <input
                  checked={false}
                  className="cjpCUo"
                  disabled={false}
                  onChange={[Function]}
                  type="checkbox"
                />
              </span>
              <span
                className="label-text cPjOWw"
                disabled={false}
              >
                Без пересадок
              </span>
            </label>
          </div>
          <div>
            <label
              className="eiMGFu"
              disabled={false}
            >
              <span
                checked={false}
                className="checkbox gJhUxw"
                disabled={false}
              >
                <input
                  checked={false}
                  className="cjpCUo"
                  disabled={false}
                  onChange={[Function]}
                  type="checkbox"
                />
              </span>
              <span
                className="label-text cPjOWw"
                disabled={false}
              >
                1 пересадка
              </span>
            </label>
          </div>
          <div>
            <label
              className="eiMGFu"
              disabled={false}
            >
              <span
                checked={false}
                className="checkbox gJhUxw"
                disabled={false}
              >
                <input
                  checked={false}
                  className="cjpCUo"
                  disabled={false}
                  onChange={[Function]}
                  type="checkbox"
                />
              </span>
              <span
                className="label-text cPjOWw"
                disabled={false}
              >
                2 и более
              </span>
            </label>
          </div>
        </div>
      </div>
      <div
        className="kb-collapse"
        style={undefined}
      >
        <div
          className="kb-collapse-item jczfhm fatUsE"
          style={undefined}
        >
          <div
            aria-expanded={false}
            className="kb-collapse-header"
            onClick={[Function]}
            role="tab"
          >
            <div
              className="ifoCPw"
            >
              <div
                className="bQdmdY"
              >
                Пересадки
              </div>
              <svg
                className="icon kszFRA crzUUw"
                size="xxsmall"
                style={undefined}
              >
                <use
                  xlinkHref="#kb_arrow-down"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>
</div>
`;

exports[`Storyshots Icon Medium Icon with stroke & fill 1`] = `
<div
  className="hMQnet"
>
  <svg
    className="bmlaaq"
    size="medium"
    style={undefined}
  >
    <use
      xlinkHref="#kb_airport-change"
    />
  </svg>
</div>
`;

exports[`Storyshots Icon Small Icon with stroke 1`] = `
<div
  className="hMQnet"
>
  <svg
    className="bXGNbG"
    size="small"
    style={undefined}
  >
    <use
      xlinkHref="#kb_checkbox"
    />
  </svg>
</div>
`;
